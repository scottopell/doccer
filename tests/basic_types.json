{
  "root": 1,
  "index": {
    "1": {
      "id": 1,
      "crate_id": 0,
      "name": "basic_types",
      "visibility": "public", 
      "docs": "Basic types fixture for testing doccer\n\nThis crate contains simple Rust constructs to validate\nbasic parsing and rendering functionality.",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"module": {"items": [2, 3, 4, 5, 6, 7]}}
    },
    "2": {
      "id": 2,
      "crate_id": 0,
      "name": "Person",
      "visibility": "public", 
      "docs": "A simple person struct",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"struct": {"fields": [], "impls": [8, 9]}}
    },
    "3": {
      "id": 3,
      "crate_id": 0,
      "name": "Vehicle",
      "visibility": "public", 
      "docs": "Different types of vehicles",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"enum": {"variants": [10, 11, 12]}}
    },
    "4": {
      "id": 4,
      "crate_id": 0,
      "name": "MAX_USERS",
      "visibility": "public", 
      "docs": "A simple constant",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"constant": {"type": {"primitive": "usize"}}}
    },
    "5": {
      "id": 5,
      "crate_id": 0,
      "name": "rectangle_area",
      "visibility": "public", 
      "docs": "Calculates the area of a rectangle",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"function": {"sig": {"inputs": [["width", {"primitive": "f64"}], ["height", {"primitive": "f64"}]], "output": {"primitive": "f64"}}}}
    },
    "6": {
      "id": 6,
      "crate_id": 0,
      "name": "Named",
      "visibility": "public", 
      "docs": "A trait for things that can be named",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"trait": {"items": [13]}}
    },
    "7": {
      "id": 7,
      "crate_id": 0,
      "name": null,
      "visibility": "public", 
      "docs": null,
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"impl": {"trait": {"path": "Named"}, "for": {"resolved_path": {"path": "Person"}}, "items": [14]}}
    },
    "8": {
      "id": 8,
      "crate_id": 0,
      "name": null,
      "visibility": "public", 
      "docs": null,
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"impl": {"for": {"resolved_path": {"path": "Person"}}, "items": [15, 16]}}
    },
    "9": {
      "id": 9,
      "crate_id": 0,
      "name": null,
      "visibility": "public", 
      "docs": null,
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"impl": {"is_synthetic": true, "trait": {"path": "Debug"}, "for": {"resolved_path": {"path": "Person"}}}}
    },
    "10": {
      "id": 10,
      "crate_id": 0,
      "name": "Car",
      "visibility": "public", 
      "docs": "A car with number of doors",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"variant": {"kind": {"tuple": [17]}}}
    },
    "11": {
      "id": 11,
      "crate_id": 0,
      "name": "Bike",
      "visibility": "public", 
      "docs": "A bicycle",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"variant": {"kind": "plain"}}
    },
    "12": {
      "id": 12,
      "crate_id": 0,
      "name": "Truck",
      "visibility": "public", 
      "docs": "A truck with cargo capacity in tons",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"variant": {"kind": {"struct": {"fields": [18]}}}}
    },
    "13": {
      "id": 13,
      "crate_id": 0,
      "name": "name",
      "visibility": "public", 
      "docs": "Returns the name",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"function": {"sig": {"inputs": [["self", {"borrowed_ref": {"type": {"generic": "Self"}}}]], "output": {"borrowed_ref": {"type": {"primitive": "str"}}}}}}
    },
    "14": {
      "id": 14,
      "crate_id": 0,
      "name": "name",
      "visibility": "public", 
      "docs": null,
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"function": {"sig": {"inputs": [["self", {"borrowed_ref": {"type": {"generic": "Self"}}}]], "output": {"borrowed_ref": {"type": {"primitive": "str"}}}}}}
    },
    "15": {
      "id": 15,
      "crate_id": 0,
      "name": "new",
      "visibility": "public", 
      "docs": "Creates a new person",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"function": {"sig": {"inputs": [["name", {"resolved_path": {"path": "String"}}], ["age", {"primitive": "u32"}]], "output": {"generic": "Self"}}}}
    },
    "16": {
      "id": 16,
      "crate_id": 0,
      "name": "get_name",
      "visibility": "public", 
      "docs": "Gets the person's name",
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"function": {"sig": {"inputs": [["self", {"borrowed_ref": {"type": {"generic": "Self"}}}]], "output": {"borrowed_ref": {"type": {"primitive": "str"}}}}}}
    },
    "17": {
      "id": 17,
      "crate_id": 0,
      "name": null,
      "visibility": "public", 
      "docs": null,
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"struct_field": {"primitive": "u8"}}
    },
    "18": {
      "id": 18,
      "crate_id": 0,
      "name": "capacity",
      "visibility": "public", 
      "docs": null,
      "links": {},
      "attrs": [],
      "deprecation": null,
      "inner": {"struct_field": {"primitive": "f32"}}
    }
  }
}