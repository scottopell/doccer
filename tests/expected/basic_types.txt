# Crate: basic_types

Basic types fixture for testing doccer

This crate contains simple Rust constructs to validate
basic parsing and rendering functionality.

  pub struct Person { ... }
    /// A simple person struct

      pub fn new(name: String, age: u32) -> Self
        /// Creates a new person

      pub fn get_name(&self) -> &str
        /// Gets the person's name

  pub enum Vehicle { ... }
    /// Different types of vehicles

    Car(u8)
      /// A car with number of doors

    Bike
      /// A bicycle

    Truck { capacity: f32 }
      /// A truck with cargo capacity in tons

  pub const MAX_USERS: usize
    /// A simple constant

  pub fn rectangle_area(width: f64, height: f64) -> f64
    /// Calculates the area of a rectangle

  pub trait Named { ... }
    /// A trait for things that can be named

    pub fn name(&self) -> &str
      /// Returns the name